docs/auto-tag-orders-with-a-post-office-box/script.liquid

```diff
@@ -0,0 +1,89 @@
+{% assign tag_to_add = options.tag_to_add__required %}
+{% assign send_email_notification = options.send_email_notification__boolean %}
+{% assign email_recipient = options.email_recipient__email %}
+{% assign email_subject = options.email_subject %}
+{% assign email_body = options.email_body__multiline %}
+
+{% comment %} 
+  Check address lines 1 and 2 for a post office box
+  The matching logic strips out all spaces and special characters,
+  then matches against one of several strings followed by a digit
+{% endcomment %}
+
+{% assign order_has_po_box = false %}
+{% assign po_box_regex = "(POB|POBOX|POSTBOX|POSTALBOX|POSTOFFICEBOX|BOXNUMBER|BOXNO|^BOX)\d+" %}
+
+{% assign po_box_match1
+  = order.shipping_address.address1
+  | scan: "[A-Za-z0-9]+"
+  | join: ""
+  | upcase
+  | match: po_box_regex
+%}
+{% assign po_box_match2
+  = order.shipping_address.address2
```

Comment:
I believe address2 field can be null in the webhook (at least per the Shopify dev docs). If so, then the scan will throw an error as it expects a string.

Just pop an if statement around this I think.